### PR TITLE
Adding PIPE'd Popen function

### PR DESCRIPTION
**Pull request by chrisidefix**
This new function allows arguments to be passed through `stdin` to the process started with `subprocess.Popen`. It also waits for the process to complete.

### CODE REVIEW

**Comment by chrissimpkins** [comment]
Looks great Christoph.  Thanks for submitting this.  

Any thoughts about how best to test it?

**Comment by chrisidefix** [comment]
I guess one way would be to execute a test-script that asks for input and then echos whatever it receives. I also should write into `stdin` with proper encoding - I added this for you together with a test-case - feel free to extend upon that.

**Comment by chrisidefix** [comment]
This surely needs more test-cases and my commit is more of an example.

File: tests/testfiles/pipe_input.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+read -p "" r
```
Comment (gessicagba) [inline tests/testfiles/pipe_input.sh]: Hey

**Pull request closed without merging**